Dalamud/Plugin/Services/IBuddyList.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+using Dalamud.Game.ClientState.Buddy;
+
+namespace Dalamud.Plugin.Services;
+
+/// <summary>
+/// This collection represents the buddies present in your squadron or trust party.
+/// It does not include the local player.
+/// </summary>
+public interface IBuddyList : IReadOnlyCollection<BuddyMember>
+{
+    /// <summary>
+    /// Gets the amount of battle buddies the local player has.
+    /// </summary>
+    public int Length { get; }
+    
+    /// <summary>
```

Comment:
Unsure if either of these should be part of the public API, given that they're just checking the relevant `Buddy` for `null` already.